rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -528,100 +606,172 @@ def slice_state_history(
 
     @staticmethod
     def _hash_example(
-        states: List[State], action: Text, tracker: DialogueStateTracker
+        tracker: DialogueStateTracker,
+        states: List[State],
+        labels: Optional[List[Text]] = None,
     ) -> int:
-        """Hash states for efficient deduplication."""
+        """Hash states (and optionally label) for efficient deduplication.
+        If labels is None, labels is not hashed.
+        """
+
         frozen_states = tuple(
             s if s is None else tracker.freeze_current_state(s) for s in states
         )
-        frozen_actions = (action,)
-        return hash((frozen_states, frozen_actions))
+        if labels is not None:
+            frozen_labels = tuple(labels)
+            return hash((frozen_states, frozen_labels))
+        else:
+            return hash(frozen_states)
 
     def training_states_actions_and_entities(
         self,
         trackers: List[DialogueStateTracker],
         domain: Domain,
         omit_unset_slots: bool = False,
     ) -> Tuple[List[List[State]], List[List[Text]], List[List[Dict[Text, Any]]]]:
-        """Transforms list of trackers to lists of states, actions and entity data.
+        """Transforms list of trackers to lists of states, labels and entity data.
 
         Args:
             trackers: The trackers to transform
             domain: The domain
             omit_unset_slots: If `True` do not include the initial values of slots.
 
         Returns:
-            A tuple of list of states, list of actions and list of entity data.
+            A tuple of list of states, list of labels and list of entity data.
+        """
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "'training_states_actions_and_entities' is being deprecated in "
+            "favor of 'training_states_labels_and_entities'."
+        )
+        return self.training_states_labels_and_entities(
+            trackers, domain, omit_unset_slots=omit_unset_slots
+        )
+
+    def training_states_labels_and_entities(
```

Comment:
I don't think this featurizer or intent max history featurizer unfeaturizes `action_unlikely_intent`. I'd suggest doing a similar cleanup to final states to achieve this as `_cleanup_last_user_state_with_action_listen` does.